src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,23 +1,47 @@
 import React from 'react';
+import { useAppSelector } from '../../app/store';
+import {
+  actions as filterActions,
+  filterSelector,
+} from '../../features/filter';
+import { useDispatch } from 'react-redux';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(filterSelector);
+  const dispatch = useDispatch();
+
+  const manageFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterActions.setStatusFilter(event.target.value as Status));
+  };
+
+  const manageQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setSearchQuery(event.target.value));
+  };
+
+  const manageClearSearch = () => {
+    dispatch(filterActions.resetFilters());
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select data-cy="statusSelect" onChange={manageFilterChange}>
+            <option value={Status.All}>{Status.All}</option>
+            <option value={Status.Active}>{Status.Active}</option>
+            <option value={Status.Completed}>{Status.Completed}</option>
           </select>
         </span>
       </p>
 
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
+          onChange={manageQueryChange}
+          value={filter.query}
```

Comment:
The 'value' attribute for the input should be 'filter.searchQuery' instead of 'filter.query' to match the naming convention used in the filterActions.